### PR TITLE
Config changed to easier to read format

### PR DESCRIPTION
**Pull request by killzoms**
Changed the config file to an easier to read format

```properties
# Settings can be changed here
ServerPort=11000
# Measured in milliseconds
SaveInterval=12000
MaxConnections=100
DisableConsole=False
DisableAutoSave=False
SaveName=world
ServerPassword=
AdminPassword=VDBZNOASQEZV
# Possible values: SURVIVAL, FREEDOM, HARDCORE, CREATIVE
GameMode=SURVIVAL
# Possible values: PROTOBUF, JSON
SerializerMode=PROTOBUF
# Default player stats below here
DefaultOxygenValue=45
DefaultMaxOxygenValue=45
DefaultHealthValue=80
DefaultHungerValue=50.5
DefaultThirstValue=90.5
DefaultInfectionValue=0

```
as opposed to
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <appSettings>
    <add key="MaxConnections" value="100" />
    <add key="Port" value="11000" />
    <add key="SaveInterval" value="120000" />
    <add key="GameMode" value="Survival" />
    <add key="SaveFileSerializer" value="PROTOBUF" />
    <add key="DisableConsole" value="false" />
    <add key="DisableAutoSave" value="false" />
    <add key="ServerPassword" value="" />
    <add key="AdminPassword" value="" />
    <add key="SaveName" value="world" />
    <add key="StartOxygen" value="45" />
    <add key="StartMaxOxygen" value="45" />
    <add key="StartHealth" value="80" />
    <add key="StartFood" value="50.5" />
    <add key="StartWater" value="90.5" />
    <add key="StartInfection" value="0" />
  </appSettings>
  <startup>
    <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.0" />
  </startup>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Mono.Cecil" publicKeyToken="50cebf1cceb9d05e" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-0.11.2.0" newVersion="0.11.2.0" />
      </dependentAssembly>
    </assemblyBinding>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Autofac" publicKeyToken="17863af14b0044da" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-3.5.0.0" newVersion="3.5.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>```

### CODE REVIEW

**Review by Measurity — APPROVED** [review]
Changed infection value to 0.1f to match bugfix on master.
Users will need to redo the settings but taking over the settings from the existing `NitroxServer-Subnautica.exe.config` file doesn't seem to be worth the code.

**Comment by Jannify** [comment]
I think the vars with type `ServerProperties` and name `serverConfig` are misleading and should be unified. I would tend to `ServerConfig` because its clearer.

**Comment by Measurity** [comment]
Changed the class name

File: NitroxModel/Serialization/PropertiesSerializer.cs [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            Dictionary<string, MemberInfo> typeCachedDict = GetTypeCacheDictionary<T>();
+            using StreamReader reader = new StreamReader(new FileStream(props.FileName, FileMode.Open), Encoding.UTF8);
+
+            char[] lineSeparator = { '=' };
+            int lineNum = 0;
+            string readLine;
+            while ((readLine = reader.ReadLine()) != null)
+            {
+                lineNum++;
+                if (readLine.Length < 1 || readLine[0] == '#')
```
[the remark this answers is not in the document]
Comment (Measurity) [reply NitroxModel/Serialization/PropertiesSerializer.cs:33]: Would be nice but it significantly increases parsing complexity because string values need to be quoted and have support for escaping quotes too.

```
# Should the value be "hello " or "hello"?
myprop=helllo # some comment
# Hard to parse because need to allow "some\" value" too
other="world" # my comment
```
